### PR TITLE
fix(rpc): align fee history

### PR DESCRIPTION
**Pull request by mmsqe**
Closes: #1559

## Description

- `eth_feeHistory` returns less item in `baseFeePerGas` array 
- `eth_feeHistory` returns `oldestBlock` - 1 compare with eth
- fix broken `earliest` input

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ## Unreleased
 
 ### Bug Fixes
+* (rpc) [#1613](https://github.com/evmos/ethermint/pull/1613) Change the default json-rpc listen address to localhost.
```
Comment (MalteHerrmann) [inline CHANGELOG.md:42]: This was already added in 1613
```suggestion
```

File: rpc/backend/utils.go [hunk 2]
```diff
@@ -134,7 +134,7 @@ func (b *Backend) processBlock(
 	// set basefee
 	targetOneFeeHistory.BaseFee = blockBaseFee
 	cfg := b.ChainConfig()
-	if cfg.IsLondon(big.NewInt(blockHeight)) {
+	if cfg.IsLondon(big.NewInt(blockHeight + 1)) {
```
Comment (MalteHerrmann) [inline rpc/backend/utils.go:137]: Can you give some background on where this +1 coming from?

File: tests/integration_tests/test_fee_history.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import pytest
+from web3 import Web3
+
+from .network import setup_ethermint
+from .utils import (
+    ADDRS,
+    send_transaction,
+)
```
Comment (MalteHerrmann) [inline tests/integration_tests/test_fee_history.py:8]: Nit: apparently flake8 wants this sorted differently
```suggestion
from .utils import send_transaction, ADDRS
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -39,8 +39,8 @@ Ref: https://keepachangelog.com/en/1.0.0/
 ## Unreleased
 
 ### Bug Fixes
-
```
Comment (MalteHerrmann) [inline CHANGELOG.md:42]: add this blank line back in

File: rpc/backend/utils.go (hunk 2 above)
Comment (mmsqe) [reply rpc/backend/utils.go:137]: it's to [align](https://github.com/ethereum/go-ethereum/blob/v1.10.26/eth/gasprice/feehistory.go#L91) with go-ethereum

**Review by fedekunze — APPROVED** [review]

**Comment by MalteHerrmann** [comment]
@Mergifyio backport release/v0.21.x

**Pull request merged**